### ISSUE TITLE
Service Error, icon turns to yellow exclamation mark

### ISSUE THREAD

**Issue by ijustride1** [web-scrobbler/web-scrobbler#3836]
A couple days ago, web scrobbler randomly stopped working. The icon went grey but I was able to change it to this yellow exclamation mark. When clicking on it, it says something briefly like "something went wrong, open settings to try fix it, some adblockers interfere, etc." I've tried everything to fix it, disabling my adblocker, configuring the settings, deleting and re-adding the extension, disconnecting and reconnecting my last fm account to it

**How to reproduce**
I have no idea how to reproduce the error

**Screenshots**
![image](https://github.com/web-scrobbler/web-scrobbler/assets/135016379/7dc92c54-4830-483b-a299-3df95c8d574a)

**Debug logs**
I do not know how to acquire debug logs

**Environment (please complete the following information)**
-   OS: Microsoft 10
-   Browser: Chrome
-   Extension version: 3.0.10

**Additional context**
/

**Comment by inverse** [comment]
Are you able to provide any logs? See https://github.com/web-scrobbler/web-scrobbler/wiki/Debug-the-extension

**Comment by agammell** [comment]
In the web console I'm seeing a bunch of these whenever a new song starts (with the API keys, etc properly filled in of course):

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://ws.audioscrobbler.com/2.0/?method=track.updatenowplaying&track=A+Bureaucratic+Desire+For+Extra+Capsular+Extraction&artist=Earth&sk=<sk>&album=A+Bureaucratic+Desire+For+Extra+Capsular+Extraction&duration=443&api_key=<key>&format=json&api_sig=<sig>. (Reason: CORS request did not succeed). Status code: (null).
```

Debug logs aren't anything particularly interesting. Just these messages when I switch from the extension tab to the tab in order to start playback:

```
13:53:07.449 Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://cc70da05-6a2f-be41-862e-154851796245/background/main.js:1
main.js:3:11709
```

**Comment by ijustride1** [comment]
> Are you able to provide any logs? See https://github.com/web-scrobbler/web-scrobbler/wiki/Debug-the-extension

i cant because it says to reproduce the issue, but i have no idea how to. it just kinda happened

**Comment by rgndxzzk** [comment]
for me i fixed it by adding "@@||ws.audioscrobbler.com" under my filters on ublock origin as the extension says. if u are using an adblocker try adding a whitelist to that url

**Comment by haz00** [comment]
Same logs for yandex music
> Error: Could not establish connection. Receiving end does not exist.
Adding filter helps on youtube music, but many other sites does not scrobble even with ublock is turned off.

**Comment by NeilN1** [comment]
Same issue for Apple Music:

> Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://148f6833-7aa8-42aa-9156-f663f4c951c0/background/main.js:1

SiriusXM won't even activate web scrobbler.  However, AccuRadio works fine.  

This seems to have started happening with the latest update. I noticed the Firefox toolbar icon has slightly changed.

**Comment by agammell** [comment]
> for me i fixed it by adding "@@||ws.audioscrobbler.com" under my filters on ublock origin as the extension says. if u are using an adblocker try adding a whitelist to that url

Ah ha! That did help me figure out what was going on.

For anyone else using uMatrix, you'll need to approve xhr requests for ws.audioscrobbler.com for each site you want to use it on. You can do this via the UI, or by adding it directly to the rules list: e.g. `youtube.com ws.audioscrobbler.com xhr allow`

Or you can blanket approve it for all sites: `* ws.audioscrobbler.com xhr allow`

**Comment by NeilN1** [comment]
I run Privacy Badger and DuckDuckGo Privacy Essentials. Disabled both - didn't help.

**Comment by kotlasu** [comment]
Same issue here (with music.apple.com).
Win 11 22H2, Firefox 113.0.2.
Turning off uBlock origin and Portmasted does nothing,
I've also added "@@||ws.audioscrobbler.com" and "@@||audioscrobbler.com" to uBlock origin.

Debug drops this error:
`Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://2c3c287f-ca86-477e-a3e2-4e510cb7d32e/background/main.js:1`

**Comment by menesis** [comment]
> for me i fixed it by adding "@@||ws.audioscrobbler.com" under my filters on ublock origin as the extension says. if u are using an adblocker try adding a whitelist to that url

I have added this exception rule to Brave's Global Defaults filter list and the issue went away.
Thanks

**Comment by silentzori** [comment]
I have this problem with Apple Music. 

<details>
  <summary>Debug log</summary>
  Here is the content of debug log: 


storage.Options = {
  "forceRecognize": false,
  "scrobblePodcasts": false,
  "useNotifications": true,
  "useUnrecognizedSongNotifications": true,
  "scrobbleRecognizedTracks": true,
  "scrobbleEditedTracksOnly": false,
  "scrobblePercent": 50,
  "disabledConnectors": "[hidden]"
} context-menu-aef8d6d8.js:15:50969
storage.LastFM = {
  "sessionID": "********************************",
  "sessionName": "******"
} context-menu-aef8d6d8.js:15:50969
storage.LibreFM = null context-menu-aef8d6d8.js:15:50969
storage.ListenBrainz = null context-menu-aef8d6d8.js:15:50969
storage.Maloja = null context-menu-aef8d6d8.js:15:50969
storage.Connectors = {
  "YouTube": {
    "scrobbleMusicOnly": false,
    "scrobbleEntertainmentOnly": false,
    "scrobbleMusicRecognisedOnly": true,
    "enableGetTrackInfoFromYtMusic": false
  }
} context-menu-aef8d6d8.js:15:50969
storage.ConnectorsOverrideOptions = {
  "apple-music": {
    "useNotifications": true,
    "useUnrecognizedSongNotifications": true
  }
} context-menu-aef8d6d8.js:15:50969
ListenBrainz: no data 2 context-menu-aef8d6d8.js:15:48817
No data in storage 2 context-menu-aef8d6d8.js:15:48817
storage.Options = {
  "forceRecognize": false,
  "scrobblePodcasts": false,
  "useNotifications": true,
  "useUnrecognizedSongNotifications": true,
  "scrobbleRecognizedTracks": true,
  "scrobbleEditedTracksOnly": false,
  "scrobblePercent": 50,
  "disabledConnectors": "[hidden]"
} main.js:1:11916
storage.LastFM = {
  "sessionID": "********************************",
  "sessionName": "******"
} main.js:1:11916
storage.LibreFM = null main.js:1:11916
storage.ListenBrainz = null main.js:1:11916
storage.Maloja = null main.js:1:11916
storage.Connectors = {
  "YouTube": {
    "scrobbleMusicOnly": false,
    "scrobbleEntertainmentOnly": false,
    "scrobbleMusicRecognisedOnly": true,
    "enableGetTrackInfoFromYtMusic": false
  }
} main.js:1:11916
storage.ConnectorsOverrideOptions = {
  "apple-music": {
    "useNotifications": true,
    "useUnrecognizedSongNotifications": true
  }
} main.js:1:11916
storage.Options = {
  "forceRecognize": false,
  "scrobblePodcasts": false,
  "useNotifications": true,
  "useUnrecognizedSongNotifications": true,
  "scrobbleRecognizedTracks": true,
  "scrobbleEditedTracksOnly": false,
  "scrobblePercent": 50,
  "disabledConnectors": "[hidden]"
} context-menu-aef8d6d8.js:15:50969
storage.LastFM = {
  "sessionID": "********************************",
  "sessionName": "******"
} context-menu-aef8d6d8.js:15:50969
storage.LibreFM = null context-menu-aef8d6d8.js:15:50969
storage.ListenBrainz = null context-menu-aef8d6d8.js:15:50969
storage.Maloja = null context-menu-aef8d6d8.js:15:50969
storage.Connectors = {
  "YouTube": {
    "scrobbleMusicOnly": false,
    "scrobbleEntertainmentOnly": false,
    "scrobbleMusicRecognisedOnly": true,
    "enableGetTrackInfoFromYtMusic": false
  }
} context-menu-aef8d6d8.js:15:50969
storage.ConnectorsOverrideOptions = {
  "apple-music": {
    "useNotifications": true,
    "useUnrecognizedSongNotifications": true
  }
} context-menu-aef8d6d8.js:15:50969
ListenBrainz: no data 2 context-menu-aef8d6d8.js:15:48817
No data in storage 2 context-menu-aef8d6d8.js:15:48817
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Background event page was not terminated on idle because a DevTools toolbox is attached to the extension. _generated_background_page.html
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Background event page was not terminated on idle because a DevTools toolbox is attached to the extension. _generated_background_page.html
Uncaught (in promise) ReferenceError: chromeMessage is not defined
    Ct moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:4
    fn moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:6
    m moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
    P moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
    Os moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:4:50
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
main.js:3:11709

Iterate on your code faster with the new multi-line editor mode. Use Enter to add new lines and Ctrl+Enter to run.

1

​

</details>

**Comment by silentzori** [comment]
My scrobbles from the last three days suddenly showed up on last.fm. But the song I'm currently playing doesn't seem to be scrobbling. The yellow exclamation mark appeared again. Debug log has the same error: 
Error: Could not establish connection. Receiving end does not exist.
    Ls moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js:1
[main.js:3:11709](moz-extension://54189fc8-295c-4cd7-8ab0-a8291f6eeef4/background/main.js)

**Comment by yayuyokitano** [comment]
This has become a bit of a hodgepodge of related and unrelated issues.
Looking at the list, every issue mentioned should be fixed in 3.0.11, most of them thanks to #3818
